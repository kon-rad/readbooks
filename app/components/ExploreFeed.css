.feedWrapper {
    margin-top: 60px;
}

.feedItem {
    overflow: auto;
    min-width: 80px;
}

.feedItem:hover {
    cursor: pointer;
}
