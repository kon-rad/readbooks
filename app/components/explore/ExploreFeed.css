.feed {
    margin-top: 60px;
}

.feed__item {
    overflow: auto;
    min-width: 80px;
    cursor: pointer;
}

.feed__item:hover {
    cursor: pointer;
}

.feed__item_title {
    margin-top: 0;
    margin-bottom: 4px;
}
