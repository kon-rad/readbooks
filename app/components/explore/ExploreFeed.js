// @flow

import * as React from 'react';
import type { TitleResults } from 'app/types/BooksAPI';
import './ExploreFeed.css';
import { Link } from 'react-router-dom';

type Props = {
  searchHits: Array<TitleResults>
};

class ExploreFeed extends React.Component<Props> {

  constructor(props: Props) {
    super(props);
  }

  processPostFields = (hit) => {
    let thumbnail = hit.volumeInfo.imageLinks
      ? hit.volumeInfo.imageLinks.smallThumbnail
      : '/images/no_results.svg';
    let title = hit.volumeInfo.title;
    let publishedDate = hit.volumeInfo.publishedDate;
    let authors = hit.volumeInfo.authors ? hit.volumeInfo.authors.join(', ') : '';
    let description = hit.searchInfo ? hit.searchInfo.textSnippet : '';

    return { title, publishedDate, thumbnail, authors, description };
  };

  renderSearchHit = (hit: TitleResults): React.Element<'div'> => {
    let book = this.processPostFields(hit);
    const detail = {
      pathname: '/explore/' + hit.id,
      state: book
    };

    return (
      <div key={hit.id} className='feedItem ui segment'>
        <Link to={detail}>
          <div className="content">
            <div className="ui tiny left floated image">
              <img
                src={book.thumbnail}
                alt=""
              />
            </div>
            <div className="header">{book.title}</div>
            <div className="meta">{book.publishedDate}</div>
            <div
              className="description"
              dangerouslySetInnerHTML={{
                __html: book.description
              }}
            />
          </div>
        </Link>
      </div>
    );
  };

  render() {
    return (
      <div className='feedWrapper'>
        {this.props.searchHits.map(this.renderSearchHit)}
      </div>
    );
  }
}

export default ExploreFeed;
