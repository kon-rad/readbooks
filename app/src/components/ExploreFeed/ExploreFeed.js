// @flow

import React from 'react';

type Props = {};

class ExploreFeed extends React.Component<Props>  {

  render() {
    return (
      <div>
        {/*results feed WIP*/}
      </div>
    )
  }
}

export default ExploreFeed;