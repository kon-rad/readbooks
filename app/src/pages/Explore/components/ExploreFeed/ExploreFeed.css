.feedWrapper {
    margin-top: 60px;
}

.feedItem {
    overflow: auto;
    min-width: 80px;
}

.feedItem:hover {
    cursor: pointer;
}

/*.feedItem .content .ui.tiny.image {*/
    /*min-height: 80px;*/
/*}*/