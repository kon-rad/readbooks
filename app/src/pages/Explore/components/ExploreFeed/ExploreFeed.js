// @flow

import * as React from 'react';
import type { TitleResults } from '../../../../types/BooksAPI';
import styles from './ExploreFeed.css';
import classNames from 'classnames/bind';

type Props = {
  searchHits:  Array<TitleResults>
};

class ExploreFeed extends React.Component<Props>  {

  cx = classNames.bind(styles);

  constructor(props: Props) {
    super(props);

  }

  renderSearchHit = (hit: TitleResults): React.Element<'div'> => {
    let feedItemClassNames = this.cx({
      feedItem: true,
      ui: true,
      segment: true
    });

    return(
      <div key={hit.id} className={feedItemClassNames}>
        <div className="content">
          <div className="ui tiny left floated image">
            <img src={hit.volumeInfo.imageLinks ? hit.volumeInfo.imageLinks.smallThumbnail : '/images/no_results.svg'} alt=""/>
          </div>
          <div className="header">
            {hit.volumeInfo.title}
          </div>
          <div className="meta">
            {hit.volumeInfo.publishedDate}
          </div>
          <div className="description"
               dangerouslySetInnerHTML={{__html: hit.searchInfo ? hit.searchInfo.textSnippet : null}}
               />
        </div>
      </div>
    )
  };

  render() {
    console.log(this.props.searchHits);

    return (
      <div className={styles.feedWrapper}>
        { this.props.searchHits.map(this.renderSearchHit) }
      </div>
    )
  }
}

export default ExploreFeed;