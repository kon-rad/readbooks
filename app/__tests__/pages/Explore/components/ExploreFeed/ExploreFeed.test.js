import React from 'react';
import ExploreFeed from '../../../../../src/pages/Explore/components/ExploreFeed/ExploreFeed';
import { shallow } from 'enzyme';

describe('ExploreFeed', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<ExploreFeed/>);
    expect(wrapper).toMatchSnapshot();
  })
});
