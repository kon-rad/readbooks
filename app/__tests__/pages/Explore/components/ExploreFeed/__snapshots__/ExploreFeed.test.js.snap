// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreFeed renders correctly without any results 1`] = `
<div
  className="feedWrapper"
>
  <div
    className="feedItem ui segment"
    key="_uTRAwAAQBAJ"
  >
    <div
      className="content"
    >
      <div
        className="ui tiny left floated image"
      >
        <img
          alt=""
          src="http://books.google.com/books/content?id=9gUJAQAAMAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api"
        />
      </div>
      <div
        className="header"
      >
        Practical Modern JavaScript
      </div>
      <div
        className="meta"
      >
        1995-12-01
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Many webmasters don't have time to learn a high level programming language, ",
          }
        }
      />
    </div>
  </div>
  <div
    className="feedItem ui segment"
    key="vmE4nBg4mpc"
  >
    <div
      className="content"
    >
      <div
        className="ui tiny left floated image"
      >
        <img
          alt=""
          src="http://image.com"
        />
      </div>
      <div
        className="header"
      >
        Functional JavaScript
      </div>
      <div
        className="meta"
      >
        2009-12-01
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Lorem ipsum",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`ExploreFeed renders correctly without results 1`] = `
<div
  className="feedWrapper"
>
  <div
    className="feedItem ui segment"
    key="_uTRAwAAQBAJ"
  >
    <Link
      replace={false}
      to={
        Object {
          "pathname": "/explore/_uTRAwAAQBAJ",
          "state": Object {
            "id": "_uTRAwAAQBAJ",
            "searchInfo": Object {
              "textSnippet": "Many webmasters don't have time to learn a high level programming language, ",
            },
            "volumeInfo": Object {
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=9gUJAQAAMAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
              },
              "publishedDate": "1995-12-01",
              "title": "Practical Modern JavaScript",
            },
          },
        }
      }
    >
      <div
        className="content"
      >
        <div
          className="ui tiny left floated image"
        >
          <img
            alt=""
            src="http://books.google.com/books/content?id=9gUJAQAAMAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api"
          />
        </div>
        <div
          className="header"
        >
          Practical Modern JavaScript
        </div>
        <div
          className="meta"
        >
          1995-12-01
        </div>
        <div
          className="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Many webmasters don't have time to learn a high level programming language, ",
            }
          }
        />
      </div>
    </Link>
  </div>
  <div
    className="feedItem ui segment"
    key="vmE4nBg4mpc"
  >
    <Link
      replace={false}
      to={
        Object {
          "pathname": "/explore/vmE4nBg4mpc",
          "state": Object {
            "id": "vmE4nBg4mpc",
            "searchInfo": Object {
              "textSnippet": "Lorem ipsum",
            },
            "volumeInfo": Object {
              "imageLinks": Object {
                "smallThumbnail": "http://image.com",
              },
              "publishedDate": "2009-12-01",
              "title": "Functional JavaScript",
            },
          },
        }
      }
    >
      <div
        className="content"
      >
        <div
          className="ui tiny left floated image"
        >
          <img
            alt=""
            src="http://image.com"
          />
        </div>
        <div
          className="header"
        >
          Functional JavaScript
        </div>
        <div
          className="meta"
        >
          2009-12-01
        </div>
        <div
          className="description"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Lorem ipsum",
            }
          }
        />
      </div>
    </Link>
  </div>
</div>
`;
